angular/src/app/views/settings/settings.component.ts

```diff
@@ -133,54 +169,61 @@ export class SettingsComponent {
    * @returns string
    */
   private validateSettings(): string {
-    let issueStringBuffer = "";
+    let issues = [];
 
-    if (this.stockBuffer < 0 || isNaN(this.stockBuffer)) {
-      issueStringBuffer +=
+    if (this.stockBuffer && (isNaN(this.stockBuffer) || this.stockBuffer < 0)) {
+      issues.push(
```

Comment:
L175:L181
```
let translationBuffer = this.translation.translate("buffer");
let translationNegativeNotAllowed = this.translation.translate(SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED);
issues.push(`${translationBuffer}: ${translationNegativeNotAllowed}`);
```
Personal preference, but I think the template literal string with those translations extracted out to their own variables looks a little easier to read. Otherwise not a big deal